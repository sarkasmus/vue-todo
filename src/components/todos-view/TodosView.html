<div class="todoView">

  <header class="header text-center">
    <h1>{{ date.getDay() | day }}</h1>
    <h2>{{ date.getMonth() | month }} {{ date.getDate() }}, {{ date.getFullYear() }}</h2>
  </header>

  <!-- List of todos. -->
  <div class="todos">
    <todo
      v-for="(todo, key) in todos"
      v-if="!todo.deleted"
      :key="key"
      :title="todo.title"
      :description="todo.description"
      :finished="todo.finished"
      :dueDate="todo.dueDate"

      @finished="finish(key)"
      @removed="remove(key)">
    </todo>
  </div>

  <footer>
    <button @click="create" class="btn btn-success btn-sm">Add</button>
  </footer>
</div>
