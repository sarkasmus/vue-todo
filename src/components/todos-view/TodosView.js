// @flow

import Todo from '../todo/Todo.vue'

export default {
  name: 'TodosView',
  components: {
    Todo
  },
  data () {
    return {
      date: new Date()
    }
  },
  computed: {
    todos () : Array<{title: string, description: string, finished: boolean, dueDate: string}> {
      return this.$store.getters.todos
    }
  },
  methods: {
    finish (id: number) : void {
      this.$store.dispatch('finish', id)
    },

    remove (id: number) : void {
      this.$store.dispatch('remove', id)
    }
  },
  filters: {
    day (day: number) : string {
      let days = [
        'Sunday',
        'Monday',
        'Tuesday',
        'Wednesday',
        'Thursday',
        'Friday',
        'Saturday'
      ]

      return days[day]
    },

    month (month: number) : string {
      let months = [
        'January',
        'February',
        'March',
        'April',
        'May',
        'June',
        'July',
        'August',
        'September',
        'October',
        'November',
        'December'
      ]

      return months[month]
    }
  }
}
