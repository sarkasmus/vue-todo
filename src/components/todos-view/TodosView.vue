<template src="./TodosView.html"></template>
<style src="./TodosView.scss"></style>
<script src="./TodosView.js"></script>
