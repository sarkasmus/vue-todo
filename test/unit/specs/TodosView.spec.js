import { createLocalVue, shallowMount } from '@vue/test-utils'
import Vuex from 'vuex'
import TodosView from '../../../src/components/todos-view/TodosView.vue'

const localVue = createLocalVue()
localVue.use(Vuex)

/**
 * Testing component representing single task (todo).
 */
describe('Todo', () => {
  let todosViewComponent
  let actions
  let getters
  let router

  /**
   * Inits every testcase.
   */
  beforeEach(() => {
    actions = {
      finish: jest.fn(),
      remove: jest.fn()
    }

    getters = {
      todos () {
        return {
          0: {
            title: 'Task 1',
            description: 'My awesome task',
            finished: false,
            dueDate: '2018-06-02'
          },
          3: {
            title: 'Task 2',
            description: 'My awesome task',
            finished: true,
            dueDate: '2018-06-03'
          }
        }
      }
    }

    let store = new Vuex.Store({
      actions: actions,
      getters: getters
    })

    router = {
      push: jest.fn()
    }

    todosViewComponent = shallowMount(TodosView, {
      mocks: {
        $router: router
      },
      store,
      localVue
    })
  })

  it('should be able to finish a task', () => {
    todosViewComponent.vm.finish(0)
    expect(actions.finish).toBeCalled()
  })

  it('should be able to delete a task', () => {
    todosViewComponent.vm.remove(0)
    expect(actions.remove).toBeCalled()
  })

  /**
   * Matching snapshot - HTML structure of the component.
   */
  it('should match snapshot', () => {
    expect(todosViewComponent.element).toMatchSnapshot()
  })
})
