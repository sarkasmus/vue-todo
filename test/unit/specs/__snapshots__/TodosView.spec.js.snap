// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo should match snapshot 1`] = `
<div
  class="todoView"
>
  <header
    class="header text-center"
  >
    <h1>
      Sunday
    </h1>
     
    <h2>
      June 3, 2018
    </h2>
  </header>
   
  <div
    class="todos"
  >
    <!---->
    <!---->
  </div>
   
  <footer>
    <button
      class="btn btn-success btn-sm"
    >
      Add
    </button>
  </footer>
</div>
`;
